packages/headless/src/features/query-suggest/query-suggest-slice.ts

```diff
@@ -17,7 +17,13 @@ export const querySuggestReducer = createReducer(
   (builder) =>
     builder
       .addCase(registerQuerySuggest, (state, action) => {
-        state[action.payload.id] = buildQuerySuggest(action.payload);
+        const id = action.payload.id;
+
+        if (id in state) {
+          return;
+        }
+
+        state[id] = buildQuerySuggest(action.payload);
```

Comment:
This change makes it possible to create multi-instance controllers (e.g. facets, search boxes) inside React functional components. When the functional component rerenders, the re-initialized controller will sync-up with the values in state instead of overwriting them.